vdev/src/testing/integration.rs

```diff
@@ -26,26 +30,58 @@ pub struct IntegrationTest {
     retries: u8,
 }
 
-impl IntegrationTest {
-    pub fn new(
-        integration: impl Into<String>,
+/// Integration tests are located in the `scripts/integration` dir,
+/// and are the full feature flag is `all-integration-tests`.
+pub(crate) struct IntegrationTest;
+
+impl ComposeTestT for IntegrationTest {
+    fn directory() -> &'static str {
+        INTEGRATION_TESTS_DIR
+    }
+
+    fn feature_flag() -> &'static str {
+        INTEGRATION_FEATURE_FLAG
+    }
+}
+
+/// E2E tests are located in the `scripts/e2e` dir,
+/// and are the full feature flag is `all-e2e-tests`.
+pub(crate) struct E2ETest;
+
+impl ComposeTestT for E2ETest {
+    fn directory() -> &'static str {
+        E2E_TESTS_DIR
+    }
+
+    fn feature_flag() -> &'static str {
+        E2E_FEATURE_FLAG
+    }
+}
```

Comment:
nit: It reads a _little_ odd to put the `impl` blocks for a trait above the trait itself, but that's no biggie.